### DISCUSSION TITLE
Redirects from _middleware.js with status code 308

### DISCUSSION THREAD

**Discussion by kevinsauvage** [vercel/next.js#35931] [Help] [answered]
Hello,

I use the _middleware file to add the default locale in the URL and one other redirection. 

The problem is that the status code is always 307, not good for SEO. I need to change it to 308 but I couldn't find any solution. 

Here is the code : 

```
  const { nextPathname, locale, origin, search } = req.nextUrl;

  const shouldHandleLocale =
    !PUBLIC_FILE.test(nextPathname) && !nextPathname.includes("/api/") && locale === "default";

  if (shouldHandleLocale) {
    const newRoute = `${origin}/en${nextPathname}${search}`;
    return NextResponse.redirect(newRoute);
  }

  if (shouldInjectGame(nextPathname)) {
    const path = nextPathname.split("/");
    const newRoute = `${origin}/${locale}/game/${path[path.length - 1]}`;
    return NextResponse.redirect(newRoute);
  }
```

Any idea to archieve it ? 

Thanks

**Comment by icyJoseph** [comment]
I think you could return a raw `const response = new Response(null, {status: 307})` and attach a location header to `response`, ala `response.headers.set('Location', 'somewhere')` - I might be wrong though, but I'd start with this attempt.

**Comment by kevinsauvage** [comment, marked as answer]
All good, 

I find the solution 

Just passing the status as the second argument of the redirect function makes it work.

```
 const { locale, origin, search, pathname } = req.nextUrl;

  const shouldHandleLocale =
    !PUBLIC_FILE.test(pathname) && !pathname.includes("/api/") && locale === "default";

  if (shouldHandleLocale) {
    const newRoute = `${origin}/en${pathname}${search}`;
    return NextResponse.redirect(newRoute, 308);
  }

  if (shouldInjectGame(pathname)) {
    const path = pathname.split("/");
    const newRoute = `${origin}/${locale}/game/${path[path.length - 1]}`;
    return NextResponse.redirect(newRoute, 308);
  }
```

**Reply by icyJoseph** [reply]
Good stuff. Mark it as answered if possible :) - you never know when you'll end up searching this again.
